lightclient/src/background.rs

```diff
@@ -242,6 +248,12 @@ struct ActiveSubscription {
     unsubscribe_method: String,
 }
 
+fn trim_response(response: &str) -> &str {
+    const MAX_RESPONSE_SIZE: usize = 256;
+    let len = std::cmp::min(response.len(), MAX_RESPONSE_SIZE);
+    &response[..len]
```

Comment:
nit, you could actually index into non-char boundary here because not all chars are ascii which leads to a panic..

https://github.com/paritytech/jsonrpsee/blob/master/core/src/tracing.rs#L101-#L124

Reply:
There are a couple of nightly only methods that would be perfect here, but in the absense of them one could do:

```rust
while !response.is_char_boundary(len) {
    len -= 1;
}
```

To find the appropriate place to slice a message that exceeds the length

Reply:
I've ended up with something similar to Niklas's suggestion, I found that a bit easier to follow :D